### PR TITLE
Configurable layer parameter

### PR DESCRIPTION
**Pull request by mholthausen**
## Description

This PR allows to control the parameter of a layer with the optional config parameter `requestParams`. This can be done via the admin-client interface in the data source field:

```
requestParams: {
  TRANSPARENT: true,
  FORMAT: 'image/png'
}
```

@terrestris/devs please review

## Related issues or pull requests

It is connected to https://github.com/terrestris/shogun-util/pull/64, which allows to get the `requestParam` values from `sourceConfig` field.

## Pull request type

- [x] Bugfix
- [ ] Feature
- [ ] Dependency updates
- [ ] Tests
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## Do you introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the 
  [Apache Licence Version 2.0](https://github.com/terrestris/shogun/blob/main/LICENSE).
- [x] I have followed the [guidelines for contributing](https://github.com/terrestris/shogun/blob/main/CONTRIBUTING.md).
- [x] The proposed change fits to the content of the [code of conduct](https://github.com/terrestris/shogun/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added or updated tests and documentation, and the test suite passes (run `mvn test` locally).
- [ ] I have added a screenshot/screencast to illustrate the visual output of my update.

### CODE REVIEW

File: shogun-lib/src/main/java/de/terrestris/shogun/lib/model/jsonb/layer/DefaultLayerSourceConfig.java [hunk 1]
```diff
@@ -75,5 +75,11 @@ public class DefaultLayerSourceConfig implements LayerSourceConfig {
         example = "© by terrestris GmbH & Co. KG"
     )
     private String attribution;
+
+    @Schema(
+        description = "Whether the background of the map should be drawn transparent.",
+        example = "true"
+    )
+    private String transparent;
```
[the remark this answers is not in the document]
Comment (mholthausen) [reply shogun-lib/src/main/java/de/terrestris/shogun/lib/model/jsonb/layer/DefaultLayerSourceConfig.java]: Yes, sure. Changed this to `Boolean` type.

**Comment by mholthausen** [comment]
> According to the description, the value is set to `true` by default, but actually it has no default set. Is there anything missing?

This will checked via the linked PR https://github.com/terrestris/shogun-util/pull/63 where the parameter is set to the given value or set to `true` and attached to the layer. Here it is an optional parameter. I will adjust the description to that.

File: shogun-lib/src/main/java/de/terrestris/shogun/lib/model/jsonb/layer/DefaultLayerSourceConfig.java [hunk 2]
```diff
@@ -75,5 +75,11 @@ public class DefaultLayerSourceConfig implements LayerSourceConfig {
         example = "© by terrestris GmbH & Co. KG"
     )
     private String attribution;
+
+    @Schema(
+        description = "Whether the background of the map should be drawn transparent.",
```
Comment (weskamm) [inline shogun-lib/src/main/java/de/terrestris/shogun/lib/model/jsonb/layer/DefaultLayerSourceConfig.java:80]:
```suggestion
        description = "Whether the layer should be drawn with transparency.",
```

**Review by weskamm — APPROVED** [review]
Please update description, else good to go

**Comment by mholthausen** [comment]
> Please update description, else good to go

I followed the description of the parameter in the official GeoServer documentation. But I will update this description, thanks 👍🏻

**Comment by mholthausen** [comment]
The [last change](https://github.com/terrestris/shogun/pull/540/commits/66b914cc7bb86bfda8f3100d2c59dc952a6e6d48) provides the generic field "requestParams", where free parameters can be assigned.

File: shogun-lib/src/main/java/de/terrestris/shogun/lib/model/jsonb/layer/DefaultLayerSourceConfig.java [hunk 3]
```diff
@@ -75,5 +76,11 @@ public class DefaultLayerSourceConfig implements LayerSourceConfig {
         example = "© by terrestris GmbH & Co. KG"
     )
     private String attribution;
+
+    @Schema(
+        description = "Request parameters to be passed to the service when querying a layer.",
+        example = "requestParams: {transparent: true}"
+    )
+    private Map requestParams;
```
Comment (dnlkoch) [inline shogun-lib/src/main/java/de/terrestris/shogun/lib/model/jsonb/layer/DefaultLayerSourceConfig.java:84]:
```suggestion
    private HashMap<String, Object> requestParams;
```

File: shogun-lib/src/main/resources/graphql/shogun.graphqls [hunk 4]
```diff
@@ -1485,6 +1486,7 @@ type Layer implements BaseEntity {
       private Double tileSize;
       private ArrayList<Double> tileOrigin;
       private ArrayList<Double> resolutions;
+      private Map requestParams;
```
Comment (dnlkoch) [inline shogun-lib/src/main/resources/graphql/shogun.graphqls:1489]:
```suggestion
      private HashMap<String, Object> requestParams;
```

File: shogun-lib/src/main/java/de/terrestris/shogun/lib/model/jsonb/layer/DefaultLayerSourceConfig.java [hunk 5]
```diff
@@ -75,5 +76,11 @@ public class DefaultLayerSourceConfig implements LayerSourceConfig {
         example = "© by terrestris GmbH & Co. KG"
     )
     private String attribution;
+
+    @Schema(
+        description = "Request parameters to be passed to the service when querying a layer.",
+        example = "requestParams: {transparent: true}"
```
Comment (dnlkoch) [inline shogun-lib/src/main/java/de/terrestris/shogun/lib/model/jsonb/layer/DefaultLayerSourceConfig.java:82]: Is this working in the Swagger API/Docs as expected? I would expect something like `{"transparent": true}`.

**Review by dnlkoch — APPROVED** [review]

**Pull request merged**